tests/functional/tokens/test_generate.py

```diff
@@ -0,0 +1,220 @@
+import ast
+
+from pytest_bdd import given, scenario, then, when
+from pytest_bdd.parsers import parse
+
+
+@scenario(
+    "../../features/tokens/generate.feature",
+    "Admin uses HTTP API to generate a token",
+)
+def test_admin_uses_http_api_to_generate_a_token():
+    """Admin uses HTTP API to generate a token."""
+
+
+@given("the admin has the admin password", target_fixture="admin_passwd")
+def the_admin_has_the_admin_password(get_admin_pwd):
+    return get_admin_pwd
+
+
+@given(
+    "the admin gets an 'access_token' by logging in to '/api/v1/token' with a "
+    "'write:token' scope",
+    target_fixture="access_token",
+)
+def the_admin_has_generated_an_access_token_with_write_token(access_token):
+    return access_token
+
+
+@given(
+    parse("the admin adds Authorization Bearer {token} in the 'headers'"),
+    target_fixture="headers",
+)
+def the_admin_adds_authorization_token_in_the_headers(access_token, token):
+    if token == "'access_token'":
+        header_token = f"Bearer {access_token}"
+    else:
+        header_token = f"Bearer {token}"
+    headers = {"Authorization": header_token}
+    return headers
+
+
+@given(
+    parse(
+        "the admin adds JSON payload with scopes: {scopes} and "
+        "expires: {expires}"
+    ),
+    target_fixture="payload",
+)
+def the_admin_adds_payload(scopes, expires):
+    # convert string "None" to None
```

Comment:
I think you have somehow moved the comments.
This comment at line 50 we agreed it's not necessary.
The comment on line 51 should go above line 56.